src/librustc_trans/trans/base.rs

```diff
@@ -2623,28 +2609,74 @@ fn internalize_symbols(cx: &SharedCrateContext, reachable: &HashSet<&str>) {
             }
         }
     }
+}
 
+// Create a `__imp_<symbol> = &symbol` global for every public static `symbol`.
+// This is required to satisfy `dllimport` references to static data in .rlibs
+// when using MSVC linker.  We do this only for data, as linker can fix up
+// code references on its own.
+// See #26591, #27438
+fn create_imps(cx: &SharedCrateContext, _reachable: &HashSet<&str>) {
```

Comment:
I think it's a safe assumption that any static with external linkage at this point was a reachable static. Either it was taken care of during emission or `internalize_symbols` internalized all the non-reachable ones I believe